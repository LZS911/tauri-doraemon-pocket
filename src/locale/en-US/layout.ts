export default {
  header: {
    customization: {
      title: 'Theme Customization',
      themeLayoutPanel: {
        defaultType: 'Default',
        miniType: 'Mini',
        title: 'Theme Layout',
        subTitle: 'Choose a layout type',
      },
      themeModePanel: {
        title: 'Theme Mode',
        subTitle: 'Choose a theme mode',
        light: 'Light',
        dark: 'Dark',
      },
      colorSchemePanel: {
        title: 'Color Scheme',
        subTitle: 'Choose a color scheme',
        default: 'Default',
        theme1: 'Theme1',
        theme2: 'Theme2',
      },
      fontFamilyPanel: {
        title: 'Font Family',
        subTitle: 'Choose a font family',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: 'Notification',
      titleTips: 'Mark as all read',
      viewAll: 'View All',
    },
    userNavigation: {
      profile: {
        title: 'profile',
        editProfile: 'Edit Profile',
        viewProfile: 'View Profile',
        bindEmail: 'Bind Email',
        modifyPassword: 'Modify Password',
        singOut: 'SignOut',
      },
      settings: {
        title: 'settings',
        accountSetting: 'Account Setting',
        permissionSetting: 'Permission Setting',
      },
    },
  },
};
