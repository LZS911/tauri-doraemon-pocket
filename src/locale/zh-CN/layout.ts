export default {
  header: {
    customization: {
      title: '主题定制',
      themeLayoutPanel: {
        defaultType: '默认',
        miniType: '简略',
        title: '布局主题',
        subTitle: '选择布局类型',
      },
      themeModePanel: {
        title: '主题模式',
        subTitle: '选择主题类型',
        light: '明亮',
        dark: '暗色',
      },
      colorSchemePanel: {
        title: '配色方案',
        subTitle: '选择配色方案',
        default: '默认',
        theme1: '主题1',
        theme2: '主题2',
      },
      fontFamilyPanel: {
        title: '字体系列',
        subTitle: '选择字体',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: '通知',
      titleTips: '标记为全部已读',
      viewAll: '查看全部',
    },
    userNavigation: {
      profile: {
        title: '简介',
        editProfile: '编辑简介',
        viewProfile: '查看简介',
        bindEmail: '邮箱绑定',
        modifyPassword: '修改密码',
        singOut: '注销',
      },
      settings: {
        title: '设置',
        accountSetting: '账号设置',
        permissionSetting: '权限设置',
      },
    },
  },
};
