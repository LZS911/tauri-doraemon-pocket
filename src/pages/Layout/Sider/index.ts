import React from 'react';
import { I18nKey } from '../../../locale';
import Sider from './Sider';
export interface INavIconList {
  name: string;
  path: string;
  title: I18nKey;
  icon?: React.ReactNode;
  onClick?: () => void;
  children?: Array<Omit<INavIconList, 'icon' | 'children'>>;
}

export default Sider;
