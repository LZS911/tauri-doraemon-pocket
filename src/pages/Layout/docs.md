# Generate Layout page by gl-cli
