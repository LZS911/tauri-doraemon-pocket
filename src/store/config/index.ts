import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
import {
  ThemeModeEnum,
  ColorSchemeEnum,
  FontFamilyEnum,
} from '../../common/enum';
import { SupportLanguage } from '../../locale';
export interface IUserConfigState {
  currentThemeMode: ThemeModeEnum;
  currentColorScheme: ColorSchemeEnum;
  lang: SupportLanguage;
  fontFamily: FontFamilyEnum;
}

export const userConfigInitialState: IUserConfigState = {
  currentThemeMode: ThemeModeEnum.Light,
  currentColorScheme: ColorSchemeEnum.Blue,
  lang: SupportLanguage.zhCN,
  fontFamily: FontFamilyEnum.Inter,
};
const config = createSlice({
  name: 'config',
  initialState: userConfigInitialState,
  reducers: {
    setCurrentThemeMode: (state, action: PayloadAction<ThemeModeEnum>) => {
      state.currentThemeMode = action.payload;
    },
    setCurrentColorScheme: (state, action: PayloadAction<ColorSchemeEnum>) => {
      state.currentColorScheme = action.payload;
    },
    setCurrentLang: (state, action: PayloadAction<SupportLanguage>) => {
      state.lang = action.payload;
    },
    setFontFamily: (state, action: PayloadAction<FontFamilyEnum>) => {
      state.fontFamily = action.payload;
    },
  },
});
export const {
  setCurrentThemeMode,
  setCurrentColorScheme,
  setCurrentLang,
  setFontFamily,
} = config.actions;
export default config.reducer;
