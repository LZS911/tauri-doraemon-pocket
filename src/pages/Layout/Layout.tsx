import { ILayoutProps } from '.';
import ThemeBase from '../../components/ThemeBase';
import Main from './Main';
import Sider from './Sider';

const Layout: React.FC<ILayoutProps> = ({ children }) => {
  return (
    <ThemeBase.Paper className="flex min-h-full w-full">
      <Sider />
      <div className="flex-auto overflow-scroll transition-[width]">
        <Main>{children}</Main>
      </div>
    </ThemeBase.Paper>
  );
};

export default Layout;
