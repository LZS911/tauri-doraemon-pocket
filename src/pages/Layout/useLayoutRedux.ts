import { useDispatch, useSelector } from 'react-redux';
import { IReduxState } from '../../store';
import {
  setIsExpandSider as _setIsExpandSider,
  changeSiderWidth as _changeSiderWidth,
  setIsDrawerSider as _setIDrawerSider,
} from '../../store/layout';

const useLayoutRedux = () => {
  const dispatch = useDispatch();
  const reduxState = useSelector((state: IReduxState) => {
    return {
      isExpandSider: state.layout.isExpandSider,
      siderWidth: state.layout.siderWidth,
      isDrawerSider: state.layout.isDrawerSider,
    };
  });

  const setIsExpandSider = (state: boolean) => {
    dispatch(_setIsExpandSider(state));
  };

  const changeSiderWidth = (width: number) => {
    dispatch(_changeSiderWidth(width));
  };

  const setIsDrawerSider = (mode: boolean) => {
    dispatch(_setIDrawerSider(mode));
  };

  return {
    ...reduxState,
    setIsExpandSider,
    changeSiderWidth,
    setIsDrawerSider,
  };
};

export default useLayoutRedux;
