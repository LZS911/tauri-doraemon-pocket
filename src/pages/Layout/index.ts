import React from 'react';
import Layout from './Layout';
export interface ILayoutProps {
  children: React.ReactNode;
}
export default Layout;
