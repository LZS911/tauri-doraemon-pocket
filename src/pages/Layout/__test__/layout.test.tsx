import { useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import useResizeObserver from 'use-resize-observer';
import { userConfigInitialState } from '../../../store/config';
import { layoutInitialState } from '../../../store/layout';
import { renderWithRouter } from '../../../testUtils/customRender';
import Layout from '../index';
import { waitFor } from '@testing-library/react';

const children = <>children</>;

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

jest.mock('use-resize-observer', () => {
  return {
    default: jest.fn(),
  };
});

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

describe('test Layout', () => {
  const useSelectorMock: jest.Mock = useSelector as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;
  const useResizeObserverMock: jest.Mock = useResizeObserver as jest.Mock;
  const useNavigateMock: jest.Mock = useNavigate as jest.Mock;
  const dispatchMock = jest.fn();
  const navigateMock = jest.fn();

  beforeEach(() => {
    jest.useFakeTimers();
    useDispatchMock.mockImplementation(() => dispatchMock);
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        userConfig: userConfigInitialState,
        layout: layoutInitialState,
      });
    });
    useNavigateMock.mockReturnValue(navigateMock);
    useResizeObserverMock.mockImplementation(() => {
      return {
        ref: useRef(),
      };
    });
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render Layout to match snapshot', async () => {
    const { container } = renderWithRouter(<Layout>{children}</Layout>);
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(container).toMatchSnapshot();
  });
});
