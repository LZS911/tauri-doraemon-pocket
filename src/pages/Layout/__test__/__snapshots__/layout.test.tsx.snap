// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Layout should render Layout to match snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white flex min-h-full w-full"
  >
    <nav
      class="sticky top-0 h-full overflow-y-auto transition-[width]"
      style="width: 60px;"
    >
      <div
        class="bg-white dark:bg-darkMode dark:text-white h-full w-full"
      >
        <div
          class="flex flex-col"
        >
          <img
            class="my-2 mx-auto h-10 w-10 cursor-pointer"
            data-testid="logo-img"
            src="/static/images/logo.svg"
            title="logo-img"
          />
          <div
            class="p-3"
          >
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="dashboard"
            >
              <span
                aria-label="dashboard"
                class="anticon anticon-dashboard"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="dashboard"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="p-3"
          >
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="chat"
            >
              <span
                aria-label="message"
                class="anticon anticon-message"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="message"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="p-3"
          >
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="component"
            >
              <span
                aria-label="code-sandbox"
                class="anticon anticon-code-sandbox"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="code-sandbox"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M709.6 210l.4-.2h.2L512 96 313.9 209.8h-.2l.7.3L151.5 304v416L512 928l360.5-208V304l-162.9-94zM482.7 843.6L339.6 761V621.4L210 547.8V372.9l272.7 157.3v313.4zM238.2 321.5l134.7-77.8 138.9 79.7 139.1-79.9 135.2 78-273.9 158-274-158zM814 548.3l-128.8 73.1v139.1l-143.9 83V530.4L814 373.1v175.2z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex-auto transition-[width]"
    >
      <header
        class="sticky top-0 right-0 z-[1000] border-b border-stone-100 dark:border-stone-800"
      >
        <div
          class="bg-white dark:bg-darkMode dark:text-white flex w-full items-center justify-between py-2 pr-6 pl-2"
        >
          <div
            class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
            data-testid="menu-icon"
          >
            <span
              aria-label="menu-unfold"
              class="anticon anticon-menu-unfold text-[1rem]"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="menu-unfold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
                />
              </svg>
            </span>
          </div>
          <div
            class="flex items-center"
          >
            <div>
              <div
                class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-2 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
                data-testid="open-language-options"
              >
                <span
                  aria-label="translation"
                  class="anticon anticon-translation text-[1rem]"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="translation"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <defs>
                      <style />
                    </defs>
                    <path
                      d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
                    />
                    <path
                      d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-2 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="popover-switch-icon"
            >
              <span
                class="ant-badge ant-badge-status leading-[0.875rem]"
              >
                <span
                  class="dark:text-white"
                >
                  <span
                    aria-label="bell"
                    class="anticon anticon-bell text-[1rem]"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="bell"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-7 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="draw-switch-icon"
            >
              <span
                aria-label="setting"
                class="anticon anticon-setting text-[1rem]"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="setting"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show bg-slate-100 px-2 py-1 text-base dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
              data-testid="open-popover-icon"
            >
              <img
                class="h-6 w-6 rounded-[50%] bg-primary"
                src="https://joeschmoe.io/api/v1/random"
              />
              <span
                class="ml-3 font-bold"
              />
            </div>
          </div>
        </div>
      </header>
      <main
        class="min-h-[calc(100vh-50px)] overflow-auto bg-slate-100 p-6 dark:!bg-black"
      >
        children
      </main>
    </div>
  </div>
</div>
`;
